activesupport/lib/active_support/core_ext/range/sole.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Range
+  # Returns the sole item in the range. If there are no items, or more
+  # than one item, raises Enumerable::SoleItemExpectedError.
+  #
+  #   (1..1).sole   # => 1
+  #   (2..1).sole   # => Enumerable::SoleItemExpectedError: no item found
+  #   (..1).sole    # => Enumerable::SoleItemExpectedError: multiple items found
+  def sole
+    if self.begin.nil? || self.end.nil?
+      raise ActiveSupport::EnumerableCoreExt::SoleItemExpectedError, "multiple items found"
```

Comment:
Not sure if it's worth it but wanted to mention that we have an opportunity to be more specific in the message and tell that the instance is an endless array which by definition can not represent a sole item